src/coreComponents/constitutive/ConstitutiveManager.cpp

```diff
@@ -60,25 +60,42 @@ ConstitutiveManager::hangConstitutiveRelation( string const & constitutiveRelati
                                                dataRepository::Group * const parent,
                                                localIndex const numConstitutivePointsPerParentIndex ) const
 {
+  dataRepository::Group * constitutiveGroup = parent->getGroupPointer( groupKeyStruct::constitutiveModelsString() );
+  if( constitutiveGroup == nullptr )   // Why do we need to do this check? shouldn't it always be there?
+  {
+    constitutiveGroup = &parent->registerGroup( groupKeyStruct::constitutiveModelsString() ).setSizedFromParent( 1 );
+    constitutiveGroup->resize( parent->size() );
+  }
+
   ConstitutiveBase const & constitutiveRelation = getConstitutiveRelation( constitutiveRelationInstanceName );
 
   std::unique_ptr< ConstitutiveBase > material = constitutiveRelation.deliverClone( constitutiveRelationInstanceName, parent );
 
   material->allocateConstitutiveData( *parent,
                                       numConstitutivePointsPerParentIndex );
 
-  dataRepository::Group * constitutiveGroup = parent->getGroupPointer( groupKeyStruct::constitutiveModelsString() );
-  if( constitutiveGroup == nullptr )
+  std::vector<string> const subRelationNames = material->getSubRelationNames();
+
+  for( string const & subRelationName : subRelationNames )
   {
-    constitutiveGroup = &parent->registerGroup( groupKeyStruct::constitutiveModelsString() ).setSizedFromParent( 1 );
-    constitutiveGroup->resize( parent->size() );
-  }
+    ConstitutiveBase const & subRelation = getConstitutiveRelation( subRelationName );
 
+    std::unique_ptr< ConstitutiveBase > constitutiveModel = subRelation.deliverClone( subRelationName, parent );
+
+    constitutiveModel->allocateConstitutiveData( *parent,
+                                                 numConstitutivePointsPerParentIndex );
+
+    ConstitutiveBase &
+      group = constitutiveGroup->registerGroup< ConstitutiveBase >( subRelationName, std::move( constitutiveModel ) );
+      group.setSizedFromParent( 1 );
+      group.resize( constitutiveGroup->size() );
+  }
 
   ConstitutiveBase &
   rval = constitutiveGroup->registerGroup< ConstitutiveBase >( constitutiveRelationInstanceName, std::move( material ) );
   rval.setSizedFromParent( 1 );
   rval.resize( constitutiveGroup->size() );
+  // Why do we need to return this here? Couldn't this be just void?
```

Comment:
I was wondering why we need this function to return that reference. It does not seem to be used.